### PR TITLE
Handle UNSET mem/reg

### PR DESCRIPTION
**Pull request by romainthomas**
We can now create symbolic variable even if there are UNSET.

### CODE REVIEW

File: src/symbolicEngine/symbolicEngine.cpp [hunk 1]
```diff
@@ -271,18 +271,55 @@ SymbolicVariable *SymbolicEngine::convertExprToSymVar(uint64 exprId, uint64 symV
...
+{
+    SymbolicExpression    *expression = nullptr;
+    SymbolicVariable      *symVar  = nullptr;
+    std::stringstream     newExpr;
+    std::stringstream     expr("#0");
+
+    expression = this->newSymbolicExpression(expr,symVarComment);
+    this->addMemoryReference(memAddr, expression->getID());
+
+    if (expression == nullptr)
```
Comment (JonathanSalwan) [inline src/symbolicEngine/symbolicEngine.cpp]: This check must be applied before the line 282.

File: src/symbolicEngine/symbolicEngine.cpp [hunk 2]
```diff
@@ -310,10 +347,15 @@ SymbolicVariable *SymbolicEngine::convertRegToSymVar(uint64 regId, uint64 symVar
...
+  if (regSymId == UNSET){
+    std::stringstream     expr("#0");
+    expression = this->newSymbolicExpression(expr, symVarComment);
+    this->symbolicReg[regId] = expression->getID();
+
+  }else{
+    expression = this->getExpressionFromId(regSymId);
+  }
 
   if (expression == nullptr)
```
Comment (JonathanSalwan) [inline src/symbolicEngine/symbolicEngine.cpp]: This check must be applied before the line 354.

File: src/symbolicEngine/symbolicEngine.cpp [hunk 3]
```diff
@@ -310,10 +347,15 @@ SymbolicVariable *SymbolicEngine::convertRegToSymVar(uint64 regId, uint64 symVar
     throw std::runtime_error("SymbolicEngine::convertRegToSymVar() - Invalid register ID");
 
   regSymId = this->getRegSymbolicID(regId);
-  if (regSymId == UNSET)
-    throw std::runtime_error("SymbolicEngine::convertRegToSymVar() - This register ID is UNSET");
 
-  expression = this->getExpressionFromId(regSymId);
+  if (regSymId == UNSET){
+    std::stringstream     expr("#0");
```
Comment (JonathanSalwan) [inline src/symbolicEngine/symbolicEngine.cpp]: This is not the `#0` but the `symVarName` ?

File: src/symbolicEngine/symbolicEngine.cpp [hunk 4]
```diff
@@ -271,18 +271,55 @@ SymbolicVariable *SymbolicEngine::convertExprToSymVar(uint64 exprId, uint64 symV
 }
 
 
+SymbolicVariable *SymbolicEngine::convertMemByteToSymVar(uint64 memAddr, std::string symVarComment)
+{
+    SymbolicExpression    *expression = nullptr;
+    SymbolicVariable      *symVar  = nullptr;
+    std::stringstream     newExpr;
+    std::stringstream     expr("#0");
```
Comment (JonathanSalwan) [inline src/symbolicEngine/symbolicEngine.cpp]: This is not the `#0` but the `symVarName` ?

File: src/symbolicEngine/symbolicEngine.cpp (hunk 3 above)
Comment (romainthomas) [inline src/symbolicEngine/symbolicEngine.cpp]: newSymbolicExpression(std::stringstream &src) takes a source id in parameter but as we create a new symbolic variable we don't have source id so I took the #0 value.

**Comment by JonathanSalwan** [comment]
Note: We will add this feature after the merge of the new symbolic engine's design.

**Pull request closed without merging**